### ISSUE TITLE
SlevomatCodingStandard.Classes.PropertyDeclaration.NoSpaceBeforeNullabilitySymbol incorrectly reported

### ISSUE THREAD

**Issue by shadowhand** [slevomat/coding-standard#1428]
This is a weird one... 

```php
<?php

declare(strict_types=1);

namespace Acme;

class UserEvent
{
    public static function detailsUpdated(): self
    {
        return static::create(UserEventType::DetailsUpdated);
    }

    public User $user;
}
```

This code reports:

> phpcs: SlevomatCodingStandard.Classes.PropertyDeclaration.NoSpaceBeforeNullabilitySymbol: There must be exactly one space before type hint nullability symbol of property $user.

But there is nothing nullable here...

My phpcs.xml is:

```xml
<?xml version="1.0"?>
<ruleset>
  <arg name="basepath" value="."/>
  <arg name="extensions" value="php"/>
  <arg name="parallel" value="80"/>
  <arg name="cache" value=".phpcs-cache"/>
  <arg name="colors"/>

  <!-- Ignore warnings, show progress of the run and show sniff names -->
  <arg value="nps"/>

  <!-- Paths to be checked -->
  <file>src</file>
  <file>tests</file>

  <!-- Allow lots of memory -->
  <ini name="memory_limit" value="2G"/>

  <!-- Include full Doctrine Coding Standard -->
  <rule ref="Doctrine">
    <exclude name="Generic.Formatting.MultipleStatementAlignment"/>
    <exclude name="Generic.NamingConventions.ConstructorName.OldStyle"/>
    <exclude name="Squiz.Arrays.ArrayDeclaration.MultiLineNotAllowed"/>
    <exclude name="SlevomatCodingStandard.Commenting.RequireOneLineDocComment"/>
    <exclude name="SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification"/>
    <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification"/>
    <exclude name="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingTraversableTypeHintSpecification"/>
    <exclude name="SlevomatCodingStandard.TypeHints.UnionTypeHintFormat.DisallowedShortNullable"/>
    <exclude name="SlevomatCodingStandard.PHP.RequireExplicitAssertion.RequiredExplicitAssertion"/>
    <exclude name="SlevomatCodingStandard.PHP.RequireNowdoc"/>
  </rule>

  <!-- Configure return type spacing -->
  <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing">
    <properties>
      <property name="spacesCountBeforeColon" value="0"/>
    </properties>
  </rule>

  <!-- Define unions style -->
  <rule ref="SlevomatCodingStandard.TypeHints.UnionTypeHintFormat">
    <properties>
      <property name="withSpaces" value="no" />
      <property name="shortNullable" value="yes" />
      <property name="nullPosition" value="last" />
    </properties>
  </rule>
</ruleset>
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/82233d918e1ee1c9487db202ce56a086ef79d5a5
